### PR TITLE
radosgw: usage: fix bytes_sent bug.

### PR DESCRIPTION
**Pull request by mdw-at-linuxbox**
http://tracker.ceph.com/issues/19870

The key parts of this fix are:
    implement AccountingFilter::complete_request() - account for bytes reported here.
    BufferingFilter<T>::complete_request() - ignore counts from send_content_length() complete_header();

There is also some debugging stuff here.  Some of it is slightly useful.  But the big thing here is passing in "cct" to more things.  I have an ugly fix here.  A better fix would be useful, although not directly relevant to this bug.

### CODE REVIEW

File: src/rgw/rgw_client_io_filters.h [hunk 1]
```diff
@@ -182,8 +226,16 @@ size_t BufferingFilter<T>::complete_request()
...
+    /* It is not correct to count these bytes here,
+     * because they can only be part of the header.
+     * Therefore force count to 0.
+     */
     sent += DecoratedRestfulClient<T>::send_content_length(data.length());
     sent += DecoratedRestfulClient<T>::complete_header();
+    ldout(cx, 30) <<
+        "BufferingFilter::complete_request: !has_content_length: IGNORE: sent="
+        << sent << dendl;
+    sent = 0;
```
Comment (rzarzynski) [inline src/rgw/rgw_client_io_filters.h]: @mdw-at-linuxbox: maybe another solution exists. Can we assume the relevant part of HTTP response, from the accounting POV, is *always* body?

File: src/rgw/rgw_client_io_filters.h [hunk 2]
```diff
@@ -14,26 +14,31 @@
 namespace rgw {
 namespace io {
 
+#define dout_subsys ceph_subsys_rgw
```
Comment (mattbenjamin) [inline src/rgw/rgw_client_io_filters.h]: we can just use lsubdout, I'm pretty sure

File: src/rgw/rgw_client_io_filters.h (hunk 1 above)
Comment (rzarzynski) [inline src/rgw/rgw_client_io_filters.h]: I'm researching this possibility. The idea is to simplify the accounting in a way that takes care only of the existing use cases (HTTP Body accounting), and thus allows to strip the extra complexity we have at the moment.

I think we still should merge the patch as a stop-gap solution.

**Comment by mdw-at-linuxbox** [comment]
I've merged PR # 16385 into this one -that has the radosgw_admin.py test logic.  I've also pulled out logic that had gone away in master.  Good riddance - I didn't want to test that part anyways.

I've made various fixes to address the issues people raised above.  I believe they're all addressed.  I made and installed one build and ran radosgw_admin.py against it - it passes.

**Comment by mattbenjamin** [comment]
@mdw-at-linuxbox , @rzarzynski I'll do a suite run

**Comment by mattbenjamin** [comment]
http://pulpito.ceph.com/mbenjamin-2017-08-09_11:31:15-rgw-wip-mdw-usage---basic-smithi/

**Pull request merged**